core/src/main/java/amino/run/policy/DefaultPolicy.java

```diff
@@ -158,6 +158,24 @@ public void onNotification(Notification notification) throws RemoteException {
                 /* Store these metrics to make them available for statistics and decision making module */
                 updateMetric(object.replicaId, object.metrics);
                 return;
+            } else if (notification instanceof MigrationNotification) {
```

Comment:
I still don't understand why we're using notifications for all of these things. 
Surely we should just have a method to receive metrics, and another one to receive migration instructions?
What's the benefit of using Notifications instead?